### ISSUE TITLE
Add support for TiDB

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#10502]
TiDB could be supported by jOOQ:
https://docs.pingcap.com/tidb/stable/overview

**Comment by Mini256** [comment]
Any things updated here? 😄

**Comment by hawkingrei** [comment]
@lukaseder I will help to complete it as soon as possible.

**Comment by hawkingrei** [comment]
@lukaseder Have you any document on how to add a new dialect for jooq？

**Comment by lukaseder** [comment]
Thanks for your suggestion, @hawkingrei. You can obviously fork jOOQ and try your luck on your own but it is *very* unlikely we'll merge your contribution upstream. For an overview on our stance to user contributions, see:
https://github.com/jOOQ/jOOQ/blob/main/CONTRIBUTING.md

In short:

- Your work is 2%, ours will still be 98%
  - You will need a lot of help getting it right
  - We'll still have to complete a lot of work ourselves, and I don't see why we should, without having sponsors cover our costs
  - You don't have access to the integration tests, and we won't give this access (see link above)
- We currently have no interest in prioritising TIDB support in jOOQ, compared to other dialects, unless we can find financial sponsorship either by the community or the DBMS vendor. There are currently 2 other dialects in the pipeline where interested parties have been open to sponsorship, so those dialects are likely to be integrated before this one.

In other words, please don't try contributing a dialect.
